### DISCUSSION TITLE
Callback function or event to intercept a write request on server

### DISCUSSION THREAD

**Discussion by sravagnan** [FreeOpcUa/python-opcua#1528] [General]
I created a server instance that has a writable boolean variable when I write the variable value from an opcua client I need to intercept the write request on the opcua server, in order to trigger a custom function that handles this event. 

Basically, I would need to intercept an event that allows me to reuse the node id and value of the data variable that is about to be written.

Is there anything already implemented that allows me to do this?

Thank you in advance.

**Comment by schroeder-** [comment]
Sry for the late response see this Pr #1445

```Python
def callback(nodeid, attr):
    return ua.DataValue(Value=ua.Variant(Value=5, VariantType=ua.VariantType.Int64))

server.set_attribute_value_callback(node.nodeid, callback, attr=ua.AttributeIds.Value)
```
